Mathlib/Probability/Moments.lean

```diff
@@ -333,4 +344,363 @@ theorem measure_le_le_exp_cgf [IsFiniteMeasure μ] (ε : ℝ) (ht : t ≤ 0)
+        suffices rexp (x * X ω) ≤
+          rexp ((2 * |t| + 1) * |X ω|) from mul_le_mul_of_nonneg_right this (sq_nonneg (X ω))
+        simp only [exp_le_exp]
+        calc
+        _ ≤ |x * X ω| := le_abs_self (x * X ω)
+        _ = |x| * |X ω| := abs_mul x (X ω)
+        _ ≤ (2 * |t| + 1) * |X ω| := by
+          suffices |x| ≤ 2 * |t| + 1 from mul_le_mul_of_nonneg_right this (abs_nonneg (X ω))
+          simp only [Metric.mem_ball] at h
+          linarith [le_trans' (le_of_lt h) (abs_sub_abs_le_abs_sub x t)]
+    exact h'
+  · apply MeasureTheory.Integrable.bdd_mul'
+    · simp only [sq_abs]
+      rw [(by ext ω; ring : (fun ω ↦ X ω ^ 2) = (fun ω ↦ X ω * X ω))]
+      apply MeasureTheory.Integrable.bdd_mul'
+        (integrable_bounded a b X hX h) (aestronglyMeasurable_iff_aemeasurable.mpr hX)
+      · filter_upwards [ha, hb] with x ha hb
+        exact (by simp only [norm_eq_abs]; exact
+          le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb) : ‖X x‖ ≤ |c|)
+    · apply aemeasurable_expt (fun ω ↦ |X ω|) (2 * |t| + 1) _
+      apply Measurable.comp_aemeasurable' _ hX
+      apply measurable_norm
+    · filter_upwards [ha, hb] with ω ha hb
+      have p0 : ‖rexp ((2 * |t| + 1) * |X ω|)‖ ≤ ‖rexp ((2 * |t| + 1) * c)‖ := by
+        simp only [norm_eq_abs, abs_exp, exp_le_exp]
+        rw [← (abs_eq_self.mpr (le_trans' (le_max_left |a| ‖b‖) (abs_nonneg a)) : |c| = c)]
+        exact mul_le_mul_of_nonneg_left
+          (le_trans' (le_abs_self (max ‖a‖ ‖b‖)) (abs_le_max_abs_abs ha hb))
+          (by linarith [abs_nonneg t] : 0 ≤ 2 * |t| + 1)
+      exact p0
+  suffices ∀ ω, ∀ x ∈ Metric.ball t (|t| + 1),
+    HasDerivAt (fun x ↦ e x ω) (e' x ω) x from ae_of_all μ this
+  intros ω x _
+  dsimp only [e, e']
+  suffices HasDerivAt (fun x ↦ rexp (x * X ω)) (rexp (x * X ω) * X ω) x from by
+    rw [(by ring : rexp (x * X ω) * X ω ^ 2 = (rexp (x * X ω) * X ω) * X ω)]
+    exact HasDerivAt.mul_const this (X ω)
+  exact HasDerivAt.exp (hasDerivAt_mul_const (X ω))
+
+theorem integrable_deriv_expt [IsFiniteMeasure μ] (t a b : ℝ)
+    (X : Ω → ℝ) (hX : AEMeasurable X μ) (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b):
+    Integrable (fun ω ↦ rexp (t * X ω) * X ω) μ := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  have hb : ∀ᵐ ω ∂μ, X ω ≤ b := h.mono fun ω h ↦ h.2
+  apply MeasureTheory.Integrable.bdd_mul'
+    (integrable_bounded a b X hX h) (aemeasurable_expt _ _ hX)
+  · set c := max ‖a‖ ‖b‖
+    filter_upwards [ha, hb] with ω ha hb
+    have q0 : ‖rexp (t * X ω)‖ ≤ ‖rexp (|t| * c)‖ := by
+      simp only [norm_eq_abs, abs_exp, exp_le_exp]
+      rw [← (abs_eq_self.mpr (le_trans' (le_max_left |a| ‖b‖) (abs_nonneg a)) : |c| = c)]
+      calc
+      _ ≤ |t * X ω| := le_abs_self (t * X ω)
+      _ = |t| * |X ω| := abs_mul t (X ω)
+      _ ≤ |t| * |c| :=
+        mul_le_mul_of_nonneg_left (le_trans' (le_abs_self (max ‖a‖ ‖b‖))
+          (abs_le_max_abs_abs ha hb) : |X ω| ≤ |c|) (abs_nonneg t)
+    exact q0
```

Comment:
The code is shorter while being just as readable if you don't split `h` into `ha` and `hb`. Given that you split `h` like that everywhere, I guess that it's to reuse the proofs that you had when the two hypotheses were separated in the statements. Please adapt the code to the new hypothesis shape.
And I suppose that you unfold `mgf` and `cgf` for the same reason: please don't, where possible. It's better to not unfold the definition unless needed, as it allows us to spot where a generic property of those definitions should be used instead of an ad-hoc computation.

```suggestion
  apply MeasureTheory.Integrable.bdd_mul'
    (integrable_bounded a b X hX h) (aemeasurable_expt _ _ hX)
  · set c := max ‖a‖ ‖b‖
    filter_upwards [h] with ω h
    have q0 : ‖rexp (t * X ω)‖ ≤ ‖rexp (|t| * c)‖ := by
      simp only [norm_eq_abs, abs_exp, exp_le_exp]
      rw [← (abs_eq_self.mpr (le_trans' (le_max_left |a| ‖b‖) (abs_nonneg a)) : |c| = c)]
      calc
      _ ≤ |t * X ω| := le_abs_self (t * X ω)
      _ = |t| * |X ω| := abs_mul t (X ω)
      _ ≤ |t| * |c| :=
        mul_le_mul_of_nonneg_left (le_trans' (le_abs_self (max ‖a‖ ‖b‖))
          (abs_le_max_abs_abs h.1 h.2) : |X ω| ≤ |c|) (abs_nonneg t)
    exact q0
```